modules/shared/src/main/scala/io/constellationnetwork/currency/dataApplication/package.scala

```diff
@@ -189,9 +193,9 @@ object BaseDataApplicationContextualOps {
       def validateData(state: DataState.Base, updates: NonEmptyList[Signed[DataUpdate]])(
         implicit context: Context
       ): F[DataApplicationValidationErrorOr[Unit]] =
-        (state.onChain, state.calculated) match {
-          case (on: DON, off: DOF) if allKnown(updates.toList) =>
-            service.validateData(DataState(on, off), updates.asInstanceOf[NonEmptyList[Signed[D]]])
+        (state.onChain, state.calculated, state.sharedArtifacts) match {
+          case (on: DON, off: DOF, sharedArtifacts: Option[SortedSet[SharedArtifact]]) if allKnown(updates.toList) =>
```

Comment:
should it be an optional value here? I don't think so 🤔 it can be just `SortedSet[SharedArtifact]`

Reply:
👍🏼